vortex-array/src/stats/bound.rs

```diff
@@ -0,0 +1,312 @@
+use std::cmp::Ordering;
+
+use vortex_error::{VortexError, VortexResult};
+use Precision::Inexact;
+
+use crate::stats::Precision::Exact;
+use crate::stats::{Precision, StatBound};
+
+/// Interpret the value as a lower bound.
+/// These format a partial order over successively more precise bounds
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct LowerBound<T>(pub(crate) Precision<T>);
+
+impl<T> LowerBound<T> {
+    pub(crate) fn min_value(self) -> T {
+        self.0.into_value()
+    }
+}
+
+impl<T> LowerBound<T> {
+    pub fn is_exact(&self) -> bool {
+        self.0.is_exact()
+    }
+}
+
+/// The result of the fallible intersection of two bound, defined to avoid Option JoinResult mixup.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum JoinResult<T> {
```

Comment:
I still don't like this. Let's talk again Monday, but at worst we should flatten the optional.